python/pyspark/sql/connect/dataframe.py

```diff
@@ -1280,10 +1281,12 @@ def __getitem__(self, item: Union[int, str, Column, List, Tuple]) -> Union[Colum
         if isinstance(item, str):
             # Check for alias
             alias = self._get_alias()
-            if alias is not None:
-                return col(alias)
-            else:
-                return col(item)
+            if self._plan is None:
+                raise Exception("Cannot analyze on empty plan.")
```

Comment:
IIRC we have now PySpark specific exception types we should use.


```suggestion
                raise SparkConnectException("Cannot analyze on empty plan.")
```

Reply:
done